src/Compiler/Microsoft.AspNetCore.Razor.Language/test/TestFiles/IntegrationTests/ComponentDesignTimeCodeGenerationTest/RenderMode_Attribute_With_Existing_Attributes/TestComponent.codegen.cs

```diff
@@ -23,7 +23,7 @@ protected override void BuildRenderTree(global::Microsoft.AspNetCore.Components.
             __o = "";
 #nullable restore
 #line 1 "x:\dir\subdir\Test\TestComponent.cshtml"
-                                     global::Microsoft.AspNetCore.Components.IComponentRenderMode __renderMode = Microsoft.AspNetCore.Components.DefaultRenderModes.Server;
+                                     __o = (global::Microsoft.AspNetCore.Components.IComponentRenderMode)(Microsoft.AspNetCore.Components.Web.RenderMode.Server);
```

Comment:
what happens if components are rendered in parallel? The `_o` is a `static` here.

Reply:
This is effectively just a type check. If the conversion can't succeed then the diagnostic gets reported by roslyn, and mapped back onto the razor doc via the line, resulting in a squiggle at design time.